### PR TITLE
Fix 'forget node' race condition in cluster-shards tcl test

### PR DESCRIPTION
**Pull request by bentotten**
Now that we have the Forget Node cluster bus extension, there is a race condition in the cluster-shards tcl test where sometimes a node is commanded to forget a node it has already forgotten due to gossip. This fixes this problem by only forgetting a node if it is known, allowing the test to proceed.

### CODE REVIEW

File: tests/cluster/tests/28-cluster-shards.tcl [hunk 1]
```diff
@@ -204,7 +204,10 @@ test "Regression test for a crash when calling SHARDS during handshake" {
     set id [R 19 CLUSTER MYID]
     R 19 CLUSTER RESET HARD
     for {set i 0} {$i < 19} {incr i} {
-        R $i CLUSTER FORGET $id
+    	# If node has not already been forgotten, forget it
```
Comment (PingXie) [inline tests/cluster/tests/28-cluster-shards.tcl:207]: What was the failure you were getting before this change? If the gossip raced ahead the forget call, it should be a no-op as opposed to a failure.

Comment (bentotten) [reply tests/cluster/tests/28-cluster-shards.tcl:207]: Hi Ping - I saw this error:
```
05:42:41> Regression test for a crash when calling SHARDS during handshake: FAILED: caught an error in the test ERR Unknown node 25bf506776eb206aa8269a3bc96ae154e6c2ccf0
ERR Unknown node 25bf506776eb206aa8269a3bc96ae154e6c2ccf0
    while executing
"[Rn $n] {*}$args"
    (procedure "R" line 2)
    invoked from within
"R $i CLUSTER FORGET $id"
    ("uplevel" body line 6)
    invoked from within
"uplevel 1 $code"
    (procedure "test" line 6)
    invoked from within
"test "Regression test for a crash when calling SHARDS during handshake" {
    # Reset forget a node, so we can use it to establish handshaking connect..."
    (file "../tests/28-cluster-shards.tcl" line 216)
    invoked from within
"source $test "
Testing unit: 29-slot-migration-response.tcl
```

Comment (bentotten) [reply tests/cluster/tests/28-cluster-shards.tcl:207]: Hi Ping - Appologies and good catch, just realized I am running this on an outdated version and not on unstable. Canceling PR

Original error:
```
05:42:41> Regression test for a crash when calling SHARDS during handshake: FAILED: caught an error in the test ERR Unknown node 25bf506776eb206aa8269a3bc96ae154e6c2ccf0
ERR Unknown node 25bf506776eb206aa8269a3bc96ae154e6c2ccf0
    while executing
"[Rn $n] {*}$args"
    (procedure "R" line 2)
    invoked from within
"R $i CLUSTER FORGET $id"
    ("uplevel" body line 6)
    invoked from within
"uplevel 1 $code"
    (procedure "test" line 6)
    invoked from within
"test "Regression test for a crash when calling SHARDS during handshake" {
    # Reset forget a node, so we can use it to establish handshaking connect..."
    (file "../tests/28-cluster-shards.tcl" line 216)
    invoked from within
"source $test "
```

Comment (PingXie) [reply tests/cluster/tests/28-cluster-shards.tcl:207]: No worries! Thanks for confirming

**Pull request closed without merging**